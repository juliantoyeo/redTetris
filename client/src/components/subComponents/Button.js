import React from 'react';
import PropTypes from 'prop-types';
import { COLORS } from '../../constants/gameConstant';

const styles = {
	mainContainer: {
		boxSizing: 'border-box',
		width: '100%',
		height: '3vw',
		color: 'white',
		cursor: 'pointer',
		backgroundColor: `rgba(${COLORS.DARK_RED}, 1)`,
		borderRadius: '1vw',
		fontSize: '1em'
	},
	disable: {
		cursor: 'cursor',
		backgroundColor: `rgba(${COLORS.RED}, 1)`
	}
}

const Button = (props) => {
	const { type, style, onClick, text, isDisable } = props;

	const onButtonClick = () => {
		if (isDisable)
			return;
		else if (onClick)
			onClick();
	}

	return (
		<button style={{ ...styles.mainContainer, ...style, ...(isDisable ? styles.disable : {}) }} type={type} onMouseDown={onButtonClick}>{text}</button>
	)
}

Button.propTypes = {
	type: PropTypes.string,
	style: PropTypes.object,
	onClick: PropTypes.func,
	text: PropTypes.string,
	isDisable: PropTypes.bool
};

export default Button;