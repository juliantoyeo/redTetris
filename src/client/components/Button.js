import React from 'react'

const styles = {
	mainContainer: {
		boxSizing: 'border-box',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		margin: '0 auto',
		height: '5vw',
		width: '100%',
		borderRadius: '4vw',
		color: 'white',
		background: '#333',
		outline: 'none',
		cursor: 'pointer'
	}
}

const Button = (props) => {
	return (
		<div style={styles.mainContainer} onClick={props.onClick}>
			{props.text}
		</div>
	)
}

export default Button