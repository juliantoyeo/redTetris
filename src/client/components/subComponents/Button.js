import React from 'react'
import { COLORS } from '../../constants/gameConstant'

const styles = {
	mainContainer: {
		boxSizing: 'border-box',
		width: '100%',
		height: '3vw',
		borderRadius: '4vw',
		color: 'white',
		cursor: 'pointer',
		backgroundColor: `rgba(${COLORS.DARK_RED}, 1)`,
		borderRadius: '1vw',
		fontSize: '1em',
	},
	disable: {
		cursor: 'cursor',
		backgroundColor: `rgba(${COLORS.RED}, 1)`,
	}
}

const Button = (props) => {
	const { type, style, onClick, text, isDisable } = props

	const onButtonClick = () => {
		if (isDisable)
			return
		else if (onClick)
			onClick()
	}

	return (
		<button style={{ ...styles.mainContainer, ...style, ...(isDisable ? styles.disable : {}) }} type={type} onMouseDown={onButtonClick}>{text}</button>
	)
}

export default Button