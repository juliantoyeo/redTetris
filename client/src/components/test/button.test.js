import React from 'react';
import { shallow } from 'enzyme';

import Button from '../subComponents/Button';

describe('Test Button component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(<Button />);
		expect(wrapper).toMatchSnapshot();
	});
	it('Should render disabled', () => {
		const wrapper = shallow(<Button isDisable={true }/>);
		expect(wrapper).toMatchSnapshot();
	});
});