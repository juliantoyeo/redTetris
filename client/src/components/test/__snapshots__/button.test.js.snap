// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Should render disabled 1`] = `
<button
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255,179,186, 1)",
      "borderRadius": "1vw",
      "boxSizing": "border-box",
      "color": "white",
      "cursor": "cursor",
      "fontSize": "1em",
      "height": "3vw",
      "width": "100%",
    }
  }
/>
`;

exports[`Test Button component Should render properly 1`] = `
<button
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255,150,150, 1)",
      "borderRadius": "1vw",
      "boxSizing": "border-box",
      "color": "white",
      "cursor": "pointer",
      "fontSize": "1em",
      "height": "3vw",
      "width": "100%",
    }
  }
/>
`;
